### ISSUE TITLE
Action does not work when `skip-github-pull-request` is set to `true`

### ISSUE THREAD

**Issue by tamoreton** [google-github-actions/release-please-action#937]
### TL;DR

When using `release-please-action` with `skip-github-pull-request: true` on a `main` branch after a squash merge, no release is created despite the action completing successfully.

### Expected behavior

I expected `release-please-action` to create a new release and update `CHANGELOG.md` and `package.json` with a new version after a PR titled with a Conventional Commit message was squash-merged into `main`.

### Observed behavior

The action completed successfully but did not create a new release, nor did it update `CHANGELOG.md` or `package.json`. This occurred even though the squash-merged PR's title followed the Conventional Commits standard.

### Action YAML

```yaml
name: release-please

on:
  push:
    branches:
      - main

permissions:
  contents: write
  pull-requests: write

jobs:
  release-please:
    runs-on: ubuntu-latest
    steps:
      - uses: google-github-actions/release-please-action@v4
        with:
          release-type: node
          skip-github-pull-request: true
```

### Log output

```text
2024-02-05T17:18:12.7205878Z Requested labels: ubuntu-latest
2024-02-05T17:18:12.7206331Z Job defined at: ***/.github/workflows/release-please.yml@refs/heads/main
2024-02-05T17:18:12.7206560Z Waiting for a runner to pick up this job...
2024-02-05T17:18:13.4667299Z Job is waiting for a hosted runner to come online.
2024-02-05T17:18:16.3051506Z Job is about to start running on the hosted runner: GitHub Actions 454 (hosted)
2024-02-05T17:18:18.7602649Z Current runner version: '2.312.0'
2024-02-05T17:18:18.7639351Z ##[group]Operating System
2024-02-05T17:18:18.7640321Z Ubuntu
2024-02-05T17:18:18.7641013Z 22.04.3
2024-02-05T17:18:18.7641490Z LTS
2024-02-05T17:18:18.7642009Z ##[endgroup]
2024-02-05T17:18:18.7642664Z ##[group]Runner Image
2024-02-05T17:18:18.7643287Z Image: ubuntu-22.04
2024-02-05T17:18:18.7643908Z Version: 20240201.1.0
2024-02-05T17:18:18.7645520Z Included Software: https://github.com/actions/runner-images/blob/ubuntu22/20240201.1/images/ubuntu/Ubuntu2204-Readme.md
2024-02-05T17:18:18.7647688Z Image Release: https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20240201.1
2024-02-05T17:18:18.7649038Z ##[endgroup]
2024-02-05T17:18:18.7649755Z ##[group]Runner Image Provisioner
2024-02-05T17:18:18.7650440Z 2.0.341.1
2024-02-05T17:18:18.7650963Z ##[endgroup]
2024-02-05T17:18:18.7652574Z ##[group]GITHUB_TOKEN Permissions
2024-02-05T17:18:18.7655163Z Contents: write
2024-02-05T17:18:18.7655971Z Metadata: read
2024-02-05T17:18:18.7656811Z PullRequests: write
2024-02-05T17:18:18.7657634Z ##[endgroup]
2024-02-05T17:18:18.7662185Z Secret source: Actions
2024-02-05T17:18:18.7663139Z Prepare workflow directory
2024-02-05T17:18:18.8574196Z Prepare all required actions
2024-02-05T17:18:18.8811563Z Getting action download info
2024-02-05T17:18:19.1190597Z Download action repository 'google-github-actions/release-please-action@v4' (SHA:cc61a07e2da466bebbc19b3a7dd01d6aecb20d1e)
2024-02-05T17:18:19.7776290Z Complete job name: release-please
2024-02-05T17:18:19.9027906Z ##[group]Run google-github-actions/release-please-action@v4
2024-02-05T17:18:19.9029030Z with:
2024-02-05T17:18:19.9029499Z   release-type: node
2024-02-05T17:18:19.9030116Z   skip-github-pull-request: true
2024-02-05T17:18:19.9031200Z   token: ***
2024-02-05T17:18:19.9031862Z   repo-url: ***
2024-02-05T17:18:19.9032824Z   github-api-url: https://api.github.com
2024-02-05T17:18:19.9033779Z   github-graphql-url: https://api.github.com/graphql
2024-02-05T17:18:19.9034700Z   fork: false
2024-02-05T17:18:19.9035264Z   include-component-in-tag: false
2024-02-05T17:18:19.9035996Z   skip-github-release: false
2024-02-05T17:18:19.9036630Z ##[endgroup]
2024-02-05T17:18:20.5402050Z Running release-please version: 16.5.0
2024-02-05T17:18:20.8799949Z ❯ Fetching package.json from branch main
2024-02-05T17:18:21.1419912Z ✔ Looking for latest release on branch: main with prefix: ***
2024-02-05T17:18:21.1425044Z ❯ Fetching merge commits on branch main with cursor: undefined
2024-02-05T17:18:21.5069553Z ❯ Fetching releases with cursor undefined
2024-02-05T17:18:21.7046263Z ❯ found release for d2s-platform-dashboard Version {
2024-02-05T17:18:21.7046996Z   major: 0,
2024-02-05T17:18:21.7047356Z   minor: 0,
2024-02-05T17:18:21.7047710Z   patch: 1,
2024-02-05T17:18:21.7048084Z   preRelease: undefined,
2024-02-05T17:18:21.7048547Z   build: undefined
2024-02-05T17:18:21.7048922Z }
2024-02-05T17:18:21.7049403Z ❯ found 1 possible releases. [
2024-02-05T17:18:21.7049931Z   Version {
2024-02-05T17:18:21.7050280Z     major: 0,
2024-02-05T17:18:21.7050649Z     minor: 0,
2024-02-05T17:18:21.7051045Z     patch: 1,
2024-02-05T17:18:21.7051421Z     preRelease: undefined,
2024-02-05T17:18:21.7051884Z     build: undefined
2024-02-05T17:18:21.7052273Z   }
2024-02-05T17:18:21.7052566Z ]
2024-02-05T17:18:21.7053723Z ✔ Building releases
2024-02-05T17:18:21.7054217Z ✔ Building strategies by path
2024-02-05T17:18:21.7054775Z ❯ .: node
2024-02-05T17:18:21.9820399Z Cleaning up orphan processes
```

**Comment by dwalker-sabiogroup** [comment]
I can confirm the same behaviour.  I have two repositories both configured with the following job

```yaml
name: release

on:
  push:
    branches:
      - main

permissions:
  contents: write
  pull-requests: write

jobs:
  release-please:
    runs-on: ubuntu-latest
    steps:
      - uses: google-github-actions/release-please-action@v4
        with:
          release-type: simple
          skip-github-pull-request: true
```

On both repositories the release action completes successfully.

On the repository where there are only commits to the main branch without PRs CHANGELOG.md and version.txt are updated and a release is created.  

However on the second repository where there is a merge commit from the action no files are updated and no release is created

**Comment by viniciusgabrielfo** [comment]
I'm having the same behavior. When using `skip-github-pull-request` basically nothing happen, either tag or releases are not created.

```
name: release-please

on:
    push:
      branches:
        - main
        - dev
  
permissions:
    contents: write
    pull-requests: write
  
jobs:
    release:
      if: ${{ github.ref == 'refs/heads/main' }}
      runs-on: ubuntu-latest
      steps:  
        - uses: google-github-actions/release-please-action@v4
          with:
            target-branch: ${{ github.ref_name }}
            config-file: release-please-config.json
    
    pre-release:
      if: ${{ github.ref == 'refs/heads/dev' }}
      runs-on: ubuntu-latest
      steps:  
        - uses: google-github-actions/release-please-action@v4
          id: pre-release
          with:
            target-branch: ${{ github.ref_name }}
            config-file: release-please-dev-config.json
            skip-github-pull-request: true
```

**Comment by vpishuk** [comment]
Same issue.

**Comment by ivanov-aleksander** [comment]
Same issue

**Comment by erikpaasonen** [comment]
Still happening on release-please version 16.12.0. (Logs from OP indicate 16.5.0.)

In my case I refactored an existing workflow that previously created releases consistently. As soon as I enabled this, it stopped creating releases altogether.

Docs are not super clear on how this new v4 param is supposed to work:

> If `true`, do not attempt to create release pull requests.

Are all of us misinterpreting what "do not attempt to create release pull requests" means? I for one thought I was going to get the tags and release published directly to the repo, bypassing the PR process.

**Comment by varrol** [comment]
This action works in this way:
- manual: push to main (assuming you run it on main)
- workflow: check previous releases and create PR with updated changelog with new release
- workflow: label new PR with: `autorelease: pending`
- manual: merge release PR
- workflow: (as another push to main triggers it) - check for existing PRs with label `autorelease: pending`
  - if found - create release and tag and label closed PR as `autorelease: tagged`
  - if not found - create another release PR :)

To sum up:
- for `skip-github-pull-request` - it does nothing as it cannot create PR - you can create release PR (with commit like `chore(main): release 1.1.0`) and label it manually - then merge and probably in this case it will create tag/release
- for `skip-labeling: true` - it does not label PRs so after merging it just creates another release PR - so the only way to make it work is to add `autorelease: pending` label to this auto created PR

I guess the intention was ok - to be able to have your custom action to create release PRs or labeling them - but someone forgot to mention in docs that having a proper PR with a proper label is required for this action to auto-create release and tag.
